pumps/moesif.go

```diff
@@ -209,7 +211,35 @@ func (p *MoesifPump) Init(config interface{}) error {
 		}).Fatal("Failed to decode configuration: ", loadConfigErr)
 	}
 
-	api := moesifapi.NewAPI(p.moesifConf.ApplicationID)
+	var apiEndpoint string
+	var batchSize int
+	var eventQueueSize int
+	var timerWakeupSeconds int
+
+	if p.moesifConf.EnableBulk && len(p.moesifConf.BulkConfig) != 0 {
+
+		// Try to fetch the event queue size from the bulk config
+		if endpoint, found := p.moesifConf.BulkConfig["api_endpoint"].(string); found {
+			apiEndpoint = endpoint
+		}
+
+		// Try to fetch the event queue size from the bulk config
+		if queueSize, found := p.moesifConf.BulkConfig["event_queue_size"].(int); found {
+			eventQueueSize = queueSize
+		}
+
+		// Try to fetch the batch size from the bulk config
+		if batch, found := p.moesifConf.BulkConfig["batch_size"].(int); found {
+			batchSize = batch
+		}
+
+		// Try to fetch the timer wake up seconds from the bulk config
+		if timer, found := p.moesifConf.BulkConfig["timer_wake_up_seconds"].(int); found {
+			timerWakeupSeconds = timer
+		}
+	}
+
+	api := moesifapi.NewAPI(p.moesifConf.ApplicationID, &apiEndpoint, eventQueueSize, batchSize, timerWakeupSeconds)
```

Comment:
@keyur9 did the `moesifapi.NewAPI` change? 
How were the queueSize, batchSize, timerWakeupSeconds values handled before?

As I understand, if we don't provide any of those values, they're going to default and still modify the old behavior.

Reply:
@tbuchaillot Yes, `moesifapi.NewAPI` did changed. We added `queueSize, batchSize, timerWakeupSeconds` config options in the `moesifapi-go` latest version (1.0.6). If the values are not provided, we'll set the default values here - https://github.com/Moesif/moesifapi-go/blob/master/api.go#L117

Reply:
@keyur9 so in this case, the `enable_bulk` config option wouldn't work since it's always going to bulk right? Since it's always going to default if not values are given.

Reply:
>>As I understand, if we don't provide any of those values, they're going to default and still modify the old behavior.

@tbuchaillot main changes here is, we wanted to have to few parameters configurable so these parameters can be tweaked depending the customer environment and their load patterns etc. So As Keyur mentioned, yes we did change the moesifapi.NewAPI to take new parameters as input and these parameters have default values(equivalent to old behavior), if no bulk config is mentioned. So if we do wanna overwrite them at specific customer, we ll ask the customer to set enable_bulk and bulk_config accordingly. Hope this answers your concern, if not let me know. Thanks

Reply:
It makes sense. Thanks for clarifying guys.